File: Week2/readability/readability.c
#include <cs50.h>
#include <ctype.h>
#include <math.h>
#include <stdio.h>
#include <string.h>

// Declare function
int calc_letter(string text);
int calc_index(int letter, int word, int sentence);

int main(void)
{
    // Promt user for text

    string text = get_string(" Text: ");

    // initialization

    int sentence = 0;
    int word = 1;
    int letter = calc_letter(text);

    // calculate sentence and word

    for (int i = 0, n = strlen(text); i < n; i++)
    {
        if (text[i] == '?' || text[i] == '.' || text[i] == '!')
        {
            sentence++;
        }
        else if (text[i] == ' ')
        {
            word++;
        }
    }

    int result = calc_index(letter, word, sentence);

    // Print grade level
    if (result < 1)
    {
        printf("Before Grade 1\n");
    }
    if (result >= 1 && result <= 16)
    {
        printf("Grade %d\n", result);
    }
    else if (result > 16)
    {
        printf("Grade 16+\n");
    }
}

// Define function
int calc_letter(string text)
{
    // calculate number of letter
    int letter = 0;
    for (int i = 0, n = strlen(text); i < n; i++)
    {

        if (isalpha(text[i]))
        {
            letter++;
        }
    }
    return letter;
}

int calc_index(int letter, int word, int sentence)
{
    float L = (float) letter / word * 100;
    float S = (float) sentence / word * 100;
    int index = round(0.0588 * L - 0.296 * S - 15.8);
    return index;
}

File: Week6/readability.py
text = str(input("Text: "))

letter = 0

for i in range(len(text)):
    if text[i].isalpha():
        letter += 1

sentence = 0
word = 1

for i in range (len(text)):
    if text[i] == '?' or text[i] == '.' or text[i] == '!':
        sentence +=1
    elif text[i] == ' ':
         word += 1

L = (letter / word) * 100
S = (sentence / word) * 100
index = round(0.0588 * L - 0.296 * S - 15.8)

if index < 1:
    print(f"Before Grade 1")
elif 1 <= index <= 16:
     print(f"Grade {index}")
elif index > 16:
     print(f"Grade 16+")
